### PR TITLE
Update the zap template used for chip-tool to not forced the first le…

### PR DESCRIPTION
**Pull request by vivien-apple**
…tter of arguments to be lower cased

 #### Problem

Followup to #6873  where arguments such as `CSRNonce` ends up as `cSRNonce`. 

 #### Summary of Changes
 * Update the template so it does not enforce the first letter to be lowercased

#### Testing
 * I have runned `./scripts/tests/test_suites.sh` just to validate that argument parsing is not broken

### CODE REVIEW

**Review by msandstedt — APPROVED** [review]

File: examples/chip-tool/templates/commands.zapt [hunk 1]
```diff
@@ -186,9 +186,9 @@ public:
     {
         {{#chip_server_cluster_command_arguments}}
         {{#if (isString type)}}
-        AddArgument("{{asCamelCased label}}", &m{{asCamelCased label false}});
+        AddArgument("{{asCamelCased label false}}", &m{{asCamelCased label false}});
```
Comment (andy31415) [inline examples/chip-tool/templates/commands.zapt:189]: For future PRs we may want to look into API readability: the false argument is not obvious in purpose. Maybe something like `startUpperCamelCase` and `startLowerCamelCase` methods would be more readable, although I find those namings awkward as well. Can these be chained? like `camelCase label | uppercaseFirstLetter`

**Review by andy31415 — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Review by woody-apple — CHANGES_REQUESTED** [review]
Per the updated template, can you update the PR here?

```
#### Problem
What is being fixed?  Examples:
* Fix crash on startup
* Fixes #12345 12345 Frobnozzle is leaky (exactly like that, so GitHub will auto-close the issue).

#### Change overview
What's in this PR

#### Testing
How was this tested? (at least one bullet point required)
    • If unit tests were added, how do they cover this issue?
    • If unit tests existed, how were they fixed/modified to prevent this in future?
    • If integration tests were added, how do they verify this change?
    • If manually tested, what platforms controller and device platforms were manually tested, and how?
    • If no testing is required, why not?
```

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@vivien-apple This is still relevant, right?

**Comment by vivien-apple** [comment]
@woody-apple I have finally updated the template...

**Review by msandstedt — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**